packages/@expo/env/src/index.ts

```diff
@@ -19,6 +19,8 @@ export const KNOWN_MODES = ['development', 'test', 'production'];
 /** The environment variable name to use when marking the environment as loaded */
 export const LOADED_ENV_NAME = '__EXPO_ENV_LOADED';
 
+const overwrittenKeys = new Set<string>();
```

Comment:
Not sure what `overwrittenKeys` does here, but the `LODED_ENV_NAME` already contains the environment variables that were loaded before - so they don't get reevaluated on subsequent runs to protect the `.env*` hierarchy and dotenv expansion.

It was also stored as global to smoothen multiple minor/patch versions existing in the same project, resulting in unexpected behavior.

Reply:
Just reverting this for now, can revisit in the future